scripts/PauseAtHeight.py

```diff
@@ -157,6 +160,19 @@ def execute(self, data: list):
                                 prevLayer = data[index - i]
                                 layer = prevLayer + layer
 
+                                # Get extruder's absolute position at the
+                                # begining of the first layer redone
+                                # see https://github.com/nallath/PostProcessingPlugin/issues/55
+                                if i == redo_layers:
```

Comment:
It's probably going to need to be reset before EVERY layer that's being redone, I think, right?

Reply:
Not sure about what you mean, but my answer is probably "no".

Let's say your Gcode is as follow (2 layers being redone):

```
layer n-2
E42
layer n-1
E50
Pause
G92 E42 --> that's what the fix does
layer n-2
E42
layer n-1
E50
```

Basically my code browses the n layers being redone backwards, finds the first one to be redone, get the extruder position, and resets to that value after the pause.

Gcode with 2 layers being redone to prove it "works":

[UO_20mm_hollow_cube.gcode.txt](https://github.com/nallath/PostProcessingPlugin/files/1571898/UO_20mm_hollow_cube.gcode.txt)

Before, we had this behavior:

```
layer n-2
E42
layer n-1
E50
Pause
G92 E50 -> position was reset to last extruder position
layer n-2
E42 -> Retraction !!! (50-42)
layer n-1
E50
```